lib/blacklight/search_state.rb

```diff
@@ -132,11 +132,14 @@ def remove_query_params
     end
 
     def filters
-      @filters ||= blacklight_config.facet_fields.each_value.map do |value|
-        f = filter(value)
-
-        f if f.any?
-      end.compact
+      @filters ||= blacklight_config.facet_fields.each_value.reduce([]) do |acc, value|
+        if value.pivot
```

Comment:
I wonder if it would it make sense to push the pivot facet handling down into `FilterField` (or maybe a subclass, `PivotFilterField`?)?

Reply:
@cbeer do you mean so that `f = filter(value)` itself handles pivots?  I think I couldn't figure out how to do that in a straight forward way.

Reply:
A complicating factor (vis-a-vis filter fields) is that the pivot values represented in the URL as plain old facets, and assembled into pivots in the search builder. We could put a PivotFilterField together, but it would either need to parameterize its values differently (maybe it actually uses the key, and has a hash of values) or just be bossy about the implicated pivot facets (they're all single values that it manipulates without regard to other facets).

Reply:
There are also some changes to make to the various sanitization stuff we've been doing if a pivot facet's component fields don't need to be present as configured filters.

Reply:
@barmintor shouldn't the fact that the field is defined via configuration to be a of type pivot be all we need?  And that's already happening.

Reply:
For what it's worth, this is what it would look like: https://github.com/projectblacklight/blacklight/pull/2696 (I don't think it's a bad idea, but it's definitely an 8.x change).